### ISSUE TITLE
[Epic] MHV CTA Widget

### ISSUE THREAD

**Issue by wesrowe** [department-of-veterans-affairs/va.gov-team#95129]
## Epic Summary

On March 17, 2025, significant changes are being made to VA.gov to keep veterans on VA.gov and stop sending them to MHV Classic. In particular, CAIA is making changes to the static pages for health tools. Today, each of these pages has a sign-in widget which requires veterans to sign-in with an identity-verified credential before they are given the link to the tool. Our attempts to make changes to these "Cerner widgets" have shown that they are nearly impossible to modify. We have a chance to reduce a great deal of UX and tech debt around the old Cerner widgets. 

In discussion with CAIA, an idea emerged to create new, "simple" widgets (`cta-widget`), which CAIA can swap in to replace the old cerner widgets when they publish the page changes through Drupal on March 17. 

## Dates
- March 3, 2025 – other teams will be testing redirects and other changes surrounding the shutting down of the Appointments tool on MHV Classic
- March 7 – go/no-go meeting will be held for Milestone 1 functionality
  - nice-to-have date for widgets existing in Staging
- March 17 – Milestone 1 changes will be published using a combination of Drupal (static content) and feature toggles (tools)
  - hard deadline for widgets to 

## Details of implementation

The March 17 milestone includes the following requirements, all wrapped into one implementation
- For static pages that link to `/my-health` tools, users must sign in with a verified ID.me or Login.gov credential to see the tool link. This functionality is provided by a react widget published through Drupal 
- There are 8 such static pages (see "URLs" for full page details):
  - 1 net-new static page about MHV portal - at (roughly) va.gov/health-care/manage/health
  - 4 static pages in benefit hubs, 1 per tool (note: 2 MR static pages are being combined into 1)
  - 3 static pages within facility pages
- Any link to My HealtheVet on the last screen of a widget points to the appropriate destination under /my-health/ (no longer Classic)
- Page content that lives _outside of the alert component_ (but within widget) is removed entirely from react widget (CAIA moving content to Drupal page)
- (optional/nice-to-have) For Veterans who have health records at an Oracle Health facility, make the widget experience the simple (final screen sends them to MHV-on-VA.gov tool)

### Notes & resources
- Mikki provided a link to the [complete IA spec](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/information-architecture/ia-design-docs/mhv-tool-landing-pages.md#page-urls-and-breadcrumbs) for Milestone 1 (3/17).
  - Includes URLs for static pages (both new and revised)

### Static pages affected:
For details on all static pages, see [this section of the IA spec](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/information-architecture/ia-design-docs/mhv-tool-landing-pages.md#required-entry-points).

**Benefit hub pages**
(see IA spec above for details on URLs, including new and revised)
- About MHV portal (Net-new page, not in production)
- Appointments 
- Medications
- Secure messages
- Medical records

**Facilities pages**
- Pharmacy - `www.va.gov/[med-center-name]/pharmacy` - e.g., https://www.va.gov/kansas-city-health-care/pharmacy/
- Appointments - `www.va.gov/[med-center-name]/make-an-appointment/` - e.g., https://www.va.gov/kansas-city-health-care/make-an-appointment/
- Medical records office - `https://www.va.gov/[med-center-name]-health-care/medical-records-office/` - e.g., https://www.va.gov/minneapolis-health-care/medical-records-office/

## Implementation notes
- The needed changes are [documented in github](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/health-care/digital-health-modernization/mhv-to-va.gov/transition/benefit-hub-page-updates.md)

## Screenshots

**AFTER / FUTURE STATE**
Example:
<img width="398" alt="Image" src="https://github.com/user-attachments/assets/15bb3402-52bf-4d02-8a44-eb21beb52dcb" />

**BEFORE/CURRENT STATE**
("Cerner widget"; may be slightly outdated)

**Example: With Oracle Health facilities**
<img width="359" alt="Image" src="https://github.com/user-attachments/assets/dcd20606-c328-407b-b63c-d211dffcb0fe" />

**Without Oracle Health facilities**
<img width="431" alt="Image" src="https://github.com/user-attachments/assets/fd9e31df-eef1-4dec-a7ec-17e34f60a67d" />

**Comment by wesrowe** [comment]
Danielle Thierry led a kick-off meeting with members of CAIA, Cartograhers, and Identity. Robyn took linear notes:
>All content in widgets in React (and on all health benefits hub pages).
Want to pull content out, make slimmer, then put in Drupal
Have widgets be sign in only -> managed by Cartography
Wes qualifier -> Cartographers working on sign in widget for Supply Reorder app. Sara working on content for general purpose for sign in. But tactical perspective - clueless on general sign in widgets.
Appointment contnet -> needs to be pulled out of widget and put in Drupal page. But sign in widget is mostly set up. So Appt widget work is just deleting content.
Going to interact directly with identity codebase? Widgets have LOA1 alert. Do we anticipate have to do new engineering that's not in widgets today.
Not that Alex is aware of.
No Drupal engineering needed. All elements and content would need in Drupal.
Does everythign have to be integrated at once? Or one by one.
Joelle: spun up a general process to follow/ pilot w/ health teams.
Wes: implement feature flag in widgets? Alex doesn't know.
Changing content at same time. Will need to edit content (some as VistA, some as OH). When put in Drupal, just one content for everyone - need to generalize content. Could time generalizing content in Drupal w/ deletion of content from React. Then change React from new pattern.
Wes would prefer do content shift one by one. Execute and test - engineers struggled to test with Supply Reorder.
Start with Appointments. Appts = test case. Update content from React, publish - then in Prod. Then just sign in widget to work with. Would load content faster, bc whole page not loading, just widget.
Timeframe for CAIA work on Appts? Not sure yet.
Decision: remove dependency of waiting on CAIA. Cartography can begin work to update widgets w/o worrying about content changes, as those can happen separately.

It would be nice to have widgets updated by early January so that the experience is robust as the Jan 31 sign-in transition approaches. Danielle anticipates increased traffic, although the scale of that is not known.

We ultimately disentangled some dependencies and reduced coordination required:
- Cartographers can update the content/UX of the sign-in widgets without worrying about the surrounding content (page content is actually inside the react widget; we'll move it into Drupal later)
- We can update widgets in sequence, they don't have to roll out simultaneously (indeed, Appts has been updated already for a long time)
- Carts will start with the ____ widget, to grok the flow and testing
- The change of MHV Classic links to VA.gov links will presumably be implemented behind a feature flag, will confirm whether that would work.

**Comment by wesrowe** [comment]
We have established that the Sign-in transition team (over at Identity, PM Joelle) will be taking responsibility for this implementation. It was a whole thing, Robyn & OCTO are aware.

Closing.

**Comment by sterkenburgsara** [comment]
@wesrowe - biggest concern I have is the need to get updates made to the My VA (va.gov/my-va) page by this timeline. The previous auth experience team was very aware of this, but there is a brand new team over there now & they may be very unaware of this timeline and ask. In theory Kaitling + Chante would be keeping tabs on this behind the scenes, but can you double-check with them? // cc @mcarterA6

**Comment by radavis** [comment]
Tickets that are successfully QA'ed:
-
